### PR TITLE
Message box

### PR DESCRIPTION
**Pull request by josegomez**
This is a new implementation of message box which is fully localizeble and it also fixes the focus bug.
 
image

There are 4 sizes that can be used Small, Medium, Large and XLarge

Usability is pretty simple
```csharp
      var result= await new Views.MessageBox("Hi", "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ut orci sed eros pharetra aliquet sit amet at ipsum. Aenean ut diam nec nisi iaculis tincidunt. Mauris at felis ante. Aliquam posuere, libero et imperdiet venenatis, turpis orci luctus nisi, eget condimentum augue magna eu risus. Suspendisse ac lorem ex. Phasellus semper null", MessageBoxSize.Small, MessageBoxButtons.OKCancel).ShowDialog<MessagBoxDialogResult>(_mainWindow);
```

### CODE REVIEW

File: SQRLDotNetClientUI/Views/MainMenuView.xaml [hunk 1]
```diff
@@ -81,7 +81,7 @@
 
       <!-- Action buttons -->
       <Button Content="{loc:Localization BtnNewIdentity}" Name="btnNewIdentity" Command="{Binding OnNewIdentityClick}" Height="30" Width="130" Margin="10" Grid.Row="1" Grid.Column="0" HorizontalAlignment="Center" />
-      <Button Content="{loc:Localization BtnRekeyIdentity}" Name="btnReKeyIdentity" Command="{Binding RekeyIdentity}" Height="30" Width="130" Margin="10" Grid.Row="1" Grid.Column="1" IsEnabled="{Binding CurrentIdentityLoaded}" HorizontalAlignment="Left"/>
+      <Button Content="{loc:Localization BtnRekeyIdentity}" Name="btnReKeyIdentity" Command="{Binding RekeyIdentity}" Height="30" Width="130" Margin="10" Grid.Row="1" Grid.Column="1"  HorizontalAlignment="Left"/>
```
Comment (alexhauser) [inline SQRLDotNetClientUI/Views/MainMenuView.xaml]: I guess you've removed this for testing, shall we put it in again?

File: SQRLDotNetClientUI/ViewModels/NewIdentityViewModel.cs [hunk 2]
```diff
@@ -74,12 +74,8 @@ public async void GenerateNewIdentity()
...
             else
             {
-                var messageBoxStandardWindow = MessageBoxManager.GetMessageBoxStandardWindow(
-                    _loc.GetLocalizationValue("ErrorTitleGeneric"),
-                    _loc.GetLocalizationValue("PasswordsDontMatchErrorMessage"),
-                    ButtonEnum.Ok, Icon.Error);
-
-                await messageBoxStandardWindow.ShowDialog(_mainWindow);
+                
+                await new Views.MessageBox(_loc.GetLocalizationValue("ErrorTitleGeneric"), _loc.GetLocalizationValue("PasswordsDontMatchErrorMessage"), MessageBoxSize.Medium, MessageBoxButtons.OK, MessageBoxIcons.ERROR).ShowDialog<MessagBoxDialogResult>(_mainWindow);
```
Comment (alexhauser) [inline SQRLDotNetClientUI/ViewModels/NewIdentityViewModel.cs]: Just a matter of taste probably, but I very much prefer spreading such long function calls with a lot of parameters out over several lines. This vastly improves readability and maintainability. For this review alone, if I were to look at every single message box call, I'd have to scroll horizontally way too often just to be able to look at all the code.

So I would prefer this:

```C#
await new Views.MessageBox(
    _loc.GetLocalizationValue("ErrorTitleGeneric"), 
    _loc.GetLocalizationValue("PasswordsDontMatchErrorMessage"), 
    MessageBoxSize.Medium, MessageBoxButtons.OK, MessageBoxIcons.ERROR)
    .ShowDialog<MessagBoxDialogResult>(_mainWindow);
```
over this:

```C#
await new Views.MessageBox(_loc.GetLocalizationValue("ErrorTitleGeneric"), _loc.GetLocalizationValue("PasswordsDontMatchErrorMessage"), MessageBoxSize.Medium, MessageBoxButtons.OK, MessageBoxIcons.ERROR).ShowDialog<MessagBoxDialogResult>(_mainWindow);
```

Just sharing my thoughts here, please feel free to ignore any of this.

File: SQRLDotNetClientUI/Views/MessageBox.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        YesNo,
+        OKCancel
+    }
+
+    public enum MessageBoxIcons
+    {
+        OK,
+        ERROR,
+        WARNING
+    }
```
Comment (alexhauser) [inline SQRLDotNetClientUI/Views/MessageBox.xaml.cs]: Question would nice, too.

File: SQRLDotNetClientUI/Views/MessageBox.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+                Width = 60
+            };
+            btn.Click += (s,e) => msgWindow.Close(clickResult); 
+            panel.Add(btn);
+        }
+
+        private void Btn_Click(object sender, Avalonia.Interactivity.RoutedEventArgs e)
+        {
+            throw new NotImplementedException();
+        }
```
Comment (alexhauser) [inline SQRLDotNetClientUI/Views/MessageBox.xaml.cs]: This isn't used I guess?

**Review by alexhauser — APPROVED** [review]
Good job, @josegomez! 

One thing I've noticed is that on the screenshot you can see that the last sentence is cut off vertically. Is this because of the message box not resizing? This can be even more problematic if the translation is longer than the original text. Let's see how this turns out in practice.

File: SQRLDotNetClientUI/Views/MessageBox.xaml.cs (hunk 4 above)
Comment (josegomez) [reply SQRLDotNetClientUI/Views/MessageBox.xaml.cs]: Yup, I used ti to get the signature for the event and forgot to remove it.. Fixed Thanks!

File: SQRLDotNetClientUI/Views/MessageBox.xaml.cs (hunk 3 above)
Comment (josegomez) [reply SQRLDotNetClientUI/Views/MessageBox.xaml.cs]: Added!

File: SQRLDotNetClientUI/ViewModels/NewIdentityViewModel.cs (hunk 2 above)
Comment (josegomez) [reply SQRLDotNetClientUI/ViewModels/NewIdentityViewModel.cs]: You are right, Fixed all instances

File: SQRLDotNetClientUI/Views/MainMenuView.xaml (hunk 1 above)
Comment (josegomez) [reply SQRLDotNetClientUI/Views/MainMenuView.xaml]: Yup, fixed

**Comment by josegomez** [comment]
> Good job, @josegomez!
> 
> One thing I've noticed is that on the screenshot you can see that the last sentence is cut off vertically. Is this because of the message box not resizing? This can be even more problematic if the translation is longer than the original text. Let's see how this turns out in practice.

@alexhauser  I found a workable solution, it will now resize the height of the box automatically based on the text, but the width of the message is set by the MessageBoxSize parameter.

image

**Comment by josegomez** [comment]
I think this is ready to merge once you give it one last once ever @alexhauser

File: SQRLDotNetClientUI/Views/MessageBox.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+using System;
+using ReactiveUI;
+using Avalonia.Platform;
+
+namespace SQRLDotNetClientUI.Views
+{
+    /// <summary>
+    /// This is a class which implements a message box to be used throughout the project
+    /// it supports localization of buttons and dynamic re-sizing where allowed.
+    /// </summary>
```
Comment (alexhauser) [inline SQRLDotNetClientUI/Views/MessageBox.xaml.cs:13]: YAY for adding documentation! 😁

**Review by alexhauser — APPROVED** [review]
Nice stuff, Jose! Also, great to see that you added documentation throughout this PR. 👍
Merge at will!

**Pull request merged**